block-producer/src/test_utils/mod.rs

```diff
@@ -7,6 +10,35 @@ use winterfell::{
     Trace, TraceInfo, TraceTable, TransitionConstraintDegree,
 };
 
+/// We need to generate a new `ProvenTransaction` every time because it doesn't
+/// derive `Clone`. Doing it this way allows us to compute the `StarkProof`
+/// once, and clone it for each new `ProvenTransaction`.
+#[derive(Clone)]
+pub struct DummyProvenTxGenerator {
```

Comment:
Should we implement `Clone` on `ProvenTransaction` in `miden-base`?

Reply:
Yes, will create a PR